### PR TITLE
Feature/125635 meetup oauth api

### PR DESCRIPTION
**Pull request by lucatume**
See: https://github.com/moderntribe/event-aggregator-site/pull/440

Tickets:
* https://central.tri.be/issues/125635
* https://central.tri.be/issues/130274

This PR adds support for Meetup API OAuth v2 flow and some deprecation messages to guide the users:
* https://www.loom.com/share/9bb2c94580044b5f8b5a37765fe11127
* https://www.loom.com/share/9ce783f345a9447fbaacd517f5802d8b
* https://www.loom.com/share/371c8e0f4c7042b4a7850fcb47b1c8af

### CODE REVIEW

File: src/admin-views/aggregator/settings.php [hunk 1]
```diff
@@ -673,7 +675,7 @@
 				'Use the options below to configure your imports. Global Import Settings apply to all imports, but you can also override the global settings by adjusting the origin-specific options. Check your Event Aggregator Service Status on the %1$s.',
 				'the-events-calendar'
 			),
-			'<a href="' . Tribe__Settings::instance()->get_url( array( 'page' => 'tribe-help' ) ) . '#tribe-tribe-aggregator-status">' . esc_html__( 'Help page', 'the-events-calendar' ) . '</a>'
+			'<a href="' . Tribe__Settings::instance()->get_url( [ 'page' => 'tribe-help' ] ) . '#tribe-tribe-aggregator-status">' . esc_html__( 'Help page', 'the-events-calendar' ) . '</a>'
```
Comment (tr1b0t) [inline src/admin-views/aggregator/settings.php]: :warning: Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')'

<sub>hash: 23f3f196672682dd25591370e62cd579</sub>

File: src/admin-views/aggregator/status.php [hunk 2]
```diff
@@ -176,16 +176,17 @@
...
-            <td><?php echo esc_html( $text ); ?></td>
-            <td><?php echo $notes; ?></td>
-        </tr>
+		<tr>
+			<td class="label">
+				<img src="<?php echo esc_url( tribe_events_resource_url( 'images/aggregator/eventbrite.png' ) ); ?>" /><span><?php esc_html_e( 'Eventbrite', 'the-events-calendar' ); ?></span>
+			</td>
+			<td class="indicator <?php esc_attr_e( $indicator ); ?>"><span class="dashicons dashicons-<?php echo esc_attr( $indicator_icons[ $indicator ] ); ?>"></span></td>
+			<td><?php echo esc_html( $text ); ?></td>
+			<td><?php echo $notes; ?></td>
```
Comment (tr1b0t) [inline src/admin-views/aggregator/status.php]: :warning: Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$notes'

<sub>hash: fee42a82cc8f878be78232e6a2337314</sub>

File: src/admin-views/aggregator/status.php [hunk 3]
```diff
@@ -176,16 +176,17 @@
...
+		</tr>
 		<?php
 		// Meetup status section
+		// @TODO: Update this section
```
Comment (tr1b0t) [inline src/admin-views/aggregator/status.php]: :warning: Comment refers to a TODO task "Update this section"

<sub>hash: 0e41d89d39d8b44050e9564f27f2d8c9</sub>

File: src/admin-views/aggregator/status.php [hunk 4]
```diff
@@ -198,14 +199,14 @@
...
-            <td><?php echo esc_html( $text ); ?></td>
-            <td><?php echo $notes; // Escaping handled above ?></td>
-        </tr>
+		<tr>
+			<td class="label">
+				<img src="<?php echo esc_url( tribe_events_resource_url( 'images/aggregator/meetup.png' ) ); ?>" /><span><?php esc_html_e( 'Meetup', 'the-events-calendar' ); ?></span>
+			</td>
+			<td class="indicator <?php esc_attr_e( $indicator ); ?>"><span class="dashicons dashicons-<?php echo esc_attr( $indicator_icons[ $indicator ] ); ?>"></span></td>
+			<td><?php echo esc_html( $text ); ?></td>
+			<td><?php echo $notes; // Escaping handled above ?></td>
```
Comment (tr1b0t) [inline src/admin-views/aggregator/status.php]: :warning: Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$notes'

<sub>hash: fee42a82cc8f878be78232e6a2337314</sub>

File: src/admin-views/aggregator/settings.php (hunk 1 above)
Comment (tr1b0t) [inline src/admin-views/aggregator/settings.php]: :warning: Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')'

<sub>hash: 1d0428bc22167b3868aaf5e0f73ef563</sub>

File: src/admin-views/aggregator/status.php [hunk 5]
```diff
@@ -41,7 +41,7 @@
 			<td class="label"><?php esc_html_e( 'License Key', 'the-events-calendar' ); ?></td>
 			<td class="indicator <?php esc_attr_e( $indicator ); ?>"><span class="dashicons dashicons-<?php echo esc_attr( $indicator_icons[ $indicator ] ); ?>"></span></td>
 			<td><?php echo esc_html( $text ); ?></td>
-			<td><?php echo $notes; ?></td>
+			<td><?php echo $notes; // Escaping handled above. ?></td>
```
Comment (tr1b0t) [inline src/admin-views/aggregator/status.php]: :warning: Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$notes'

<sub>hash: 8f988e4b898610db14e916c7519e0311</sub>

File: src/admin-views/aggregator/status.php (hunk 2 above)
Comment (tr1b0t) [inline src/admin-views/aggregator/status.php]: :warning: Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$notes'

<sub>hash: 00af99bfeca64f4352b35206f1411a8a</sub>

File: src/admin-views/aggregator/status.php (hunk 3 above)
Comment (tr1b0t) [inline src/admin-views/aggregator/status.php]: :warning: Comment refers to a TODO task "Update this section"

<sub>hash: 232050efbd2e3e22d9e47f25a5b467b8</sub>

File: src/admin-views/aggregator/status.php (hunk 4 above)
Comment (tr1b0t) [inline src/admin-views/aggregator/status.php]: :warning: Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$notes'

<sub>hash: 7a379186b842d72fdc4d7074ef508116</sub>

File: src/Tribe/Aggregator/Record/Meetup.php [hunk 6]
```diff
@@ -77,4 +77,53 @@ public static function filter_event_to_preserve_fields( $event, $record ) {
 
 		return self::preserve_event_option_fields( $event );
 	}
+
+	/**
+	 * Returns the Meetup authorization token generation URL.
+	 *
+	 * @param array $args
```
Comment (bordoni) [inline src/Tribe/Aggregator/Record/Meetup.php]: Add `@since` to the docblock.

File: src/Tribe/Aggregator/Service.php [hunk 7]
```diff
@@ -29,6 +29,11 @@ class Tribe__Events__Aggregator__Service {
 	 */
 	public static $auth_transient = 'tribe_aggregator_has_eventbrite_authorized_response';
 
+	/**
+	 * @var string
+	 */
+	public static $auth_transient_meetup = 'tribe_aggregator_has_meetup_authorized_response';
```
Comment (bordoni) [inline src/Tribe/Aggregator/Service.php]: Add `@since` to the Docblock.

File: src/Tribe/Aggregator/Settings.php [hunk 8]
```diff
@@ -175,20 +176,177 @@ public function is_ea_authorized_for_eb() {
...
+	 * Hooked to current_screen, this method identifies whether or not eb credentials should be cleared
+	 *
+	 * @param WP_Screen $screen
+	 */
+	public function maybe_clear_meetup_credentials( $screen ) {
+		if ( 'tribe_events_page_tribe-common' !== $screen->base ) {
+			return;
+		}
+
+		if ( ! isset( $_GET['tab'] ) || 'addons' !== $_GET['tab'] ) {
```
Comment (bordoni) [inline src/Tribe/Aggregator/Settings.php]: Lets use `tribe_get_request_var()`

File: src/Tribe/Aggregator/Settings.php [hunk 9]
```diff
@@ -175,20 +176,177 @@ public function is_ea_authorized_for_eb() {
...
+			return;
+		}
+
+		if (
+			! (
+				isset( $_GET['action'] )
+				&& isset( $_GET['_wpnonce'] )
```
Comment (bordoni) [inline src/Tribe/Aggregator/Settings.php]: Lets use `tribe_get_request_var()`

File: src/Tribe/Aggregator/Settings.php [hunk 10]
```diff
@@ -175,20 +176,177 @@ public function is_ea_authorized_for_eb() {
...
+				&& 'disconnect-meetup' === $_GET['action']
```
Comment (bordoni) [inline src/Tribe/Aggregator/Settings.php]: Lets use `tribe_get_request_var()`

File: src/Tribe/Aggregator/Settings.php [hunk 11]
```diff
@@ -175,20 +176,177 @@ public function is_ea_authorized_for_eb() {
...
 	}
 
+	/*
+
+		=====================================================================================================
+	*/
+
+	/**
+	 * Hooked to current_screen, this method identifies whether or not eb credentials should be cleared
+	 *
```
Comment (bordoni) [inline src/Tribe/Aggregator/Settings.php]: Docblock missing `@since`

**Comment by bordoni** [comment]
Moved into B19.09, as B19.08 went out today.

**Comment by lucatume** [comment]
Ok, updates pushed and ready for another pass.

**Pull request merged**